src/api/uploads/UploadReachability.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @flow
+ * @file Helper for Box uploads reachability test
+ * @author Box
+ */
+
+import Base from '../Base';
+import LocalStore from '../../util/LocalStore';
+import { DEFAULT_HOSTNAME_UPLOAD } from '../../constants';
+import '../../util/regenerator-runtime';
+
+const CACHED_RESULTS_LOCAL_STORE_KEY = 'uploads-reachability-cached-results';
+
+class UploadsReachability extends Base {
+    localStore: LocalStore;
+
+    /**
+     * [constructor]
+     *
+     * @param {Object} [options]
+     * @param {string} [options.token] - Auth token
+     * @param {string} [options.uploadHost] - Upload host name
+     * @return {void}
+     */
+    constructor(options: Object) {
+        super(options);
+        this.localStore = new LocalStore();
+    }
+
+    /**
+     * Gets a reachable host for upload
+     */
+    async getReachableUploadHost() {
+        const cachedResults = this.localStore.getItem(CACHED_RESULTS_LOCAL_STORE_KEY) || {};
```

Comment:
Maybe `cachedHostReachability`? Reading this, I was a bit confused and thought the actual hosts were in this cache

Reply:
done